src/Servers/Kestrel/Transport.Sockets/src/SocketConnectionListener.cs

```diff
@@ -117,28 +119,33 @@ internal void Bind()
                 {
                     Debug.Assert(_listenSocket != null, "Bind must be called first.");
 
+
                     var acceptSocket = await _listenSocket.AcceptAsync(cancellationToken);
 
                     // Only apply no delay to Tcp based endpoints
+                    var noDelay = false;
+// REVIEW: should this move to inside of _contextFactory.Create??
                     if (acceptSocket.LocalEndPoint is IPEndPoint)
                     {
                         acceptSocket.NoDelay = _options.NoDelay;
+                        noDelay = true;
                     }
 
                     var setting = _settings[_settingsIndex];
 
-                    var connection = new SocketConnection(acceptSocket,
-                        _memoryPool,
-                        setting.Scheduler,
-                        _trace,
-                        setting.SocketSenderPool,
-                        setting.InputOptions,
-                        setting.OutputOptions,
-                        waitForData: _options.WaitForDataBeforeAllocatingBuffer);
+                    var connectionOptions = new SocketConnectionOptions()
+                    {
+                        DelaySocketOperations = !noDelay,
```

Comment:
`_options.NoDelay` is different from `DelaySocketOperations` which was referring to this https://github.com/dotnet/aspnetcore/pull/34639/files#diff-6c2098732ca5c5b26b7451144b6f78147bcbad8d1b444a02b568ba575e9317f3R39. This *is* confusing though. We should probably come up with a new name for the new uption. @davidfowl

Reply:
Okay, good idea to rename :)